Gordon360/Controllers/RecIM/ActivitiesController.cs

```diff
@@ -26,18 +26,23 @@ public ActivitiesController(IActivityService activityService)
         ///<summary>Gets a list of all Activities by parameter </summary>
         ///<param name="active"> Optional active parameter </param>
         ///<param name="time"> Optional time parameter </param>
+        ///<param name="registrationOpen"> Optional registration availability parameter </param>
         /// <returns>
         /// All Existing Activities 
         /// </returns>
         [HttpGet]
         [Route("")]
-        public ActionResult<IEnumerable<ActivityViewModel>> GetActivities([FromQuery] DateTime? time, bool active)
-        {   
+        public ActionResult<IEnumerable<ActivityViewModel>> GetActivities([FromQuery] DateTime? time, bool active, bool registrationOpen)
+        {
+            if (registrationOpen)
+            {
+                return Ok(_activityService.GetOpenActivities());
+            }
             if (time is null && active)
             {
                 var activeResults = _activityService.GetActivities();
                 return Ok(activeResults);
-                
+
             }
             var result = _activityService.GetActivitiesByTime(time);
```

Comment:
I would recommend doing all of this logic inside of **one** `GetActivities(...)` service method. If you use optional (nullable) filter parameters, you can simplify the operation.

Reply:
makes sense, I did notice i was doing some backflips to work out my own logic. I'll update this in the next PR (or maybe in the cleanup PR (https://github.com/gordon-cs/gordon-360-api/pull/754#issuecomment-1328256513)